### PR TITLE
Release notes for CockroachDB v22.2.0-alpha.4

### PR DESCRIPTION
**Pull request by mdlinville**
SHA: aac413c
- [x] Local build
- [x] Local link check

### CODE REVIEW

**Comment by mdlinville** [comment]
@nickvigilante @andf-crl PTAL since this is my first time handling release notes. If you don't see anything strange I'll send it for broader review.

**Review by nickvigilante — APPROVED** [review]
This is reviewed for tooling purposes.

File: _includes/releases/v22.2/v22.2.0-alpha.4.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+- Instead of always recommending to replace the first existing index with the same explicit columns, the index recommendation now considers all existing indexes in the table to decide the best one to replace. [#87174][#87174]
+- Owners of a backup schedule can now control their schedule using the supported pause, resume, drop, and alter queries. [#87600][#87600]
+- Previously, a schema could be renamed even when a child table was referenced by a view or UDF. Renaming the schema breaks the view or UDF since tables are referenced by name in views and UDFs. A check has been added that disallows renaming a schema that is referenced in views or UDFs. [#87540][#87540]
+- Users may now be granted the CHANGEFEED privilege on a table, allowing them to create changefeeds for that table even if they don't have the CONTROLCHANGEFEED role option or the SELECT privilege. Note that this still in effect gives them the ability to read the data in the table. Users with the CONTROLCHANGEFEED role option still need SELECT on each table, even if they also have CHANGEFEED. [#87887][#87887]
```
Comment (stbof) [inline _includes/releases/v22.2/v22.2.0-alpha.4.md:18]: All these privileges and roles should be in backticks.

File: _includes/releases/v22.2/v22.2.0-alpha.4.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+- When there is no insight problem detected, a message indicates that the statement was slow and how long it took to execute. [#87799][#87799]
+- The Insight Statement Detail page is linked from the Insight Statement page but no longer includes a link to itself. [#87800][#87800]
+- Added the ability to copy an index recommendation to the clipboard. [#87794][#87794]
+- Fixed the index and grant sorting on the database page to have default column and to have url match the user selection. [#87832][#87832]
+- Added Application Name to Statement overview, Transaction Overview (and their respective column selectors), Transaction Details. Updated label from "App" to "Application Name" on Statement Details page. [#87868][#87868]
+- The value for percentage of all runtime is now calculated based on all data from the time period. Previously, it was calculated based on only the filtered data. [#88027][#88027]
+- The Contention column has been renamed to Contention time. This matches other columns such as Elapsed time. [#88040][#88040]
```
Comment (stbof) [inline _includes/releases/v22.2/v22.2.0-alpha.4.md:35]:
```suggestion
- The Contention column has been renamed to Contention Time. This matches other columns such as Elapsed Time. [#88040][#88040]
```

File: _includes/releases/v22.2/v22.2.0-alpha.4.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+- Previously, the `is_generated` column in the `information_schema.column` table returned either `YES` or `NO`, depending on whether the column is computed. The column now returns either `ALWAYS` or `NEVER`. This matches the behavior of PostgreSQL. [#87670][#87670]
+- In rare cases, the value of a cluster setting could revert soon after it was updated. This no longer happens for a given gateway node. [#87732][#87732]
+- The UI no longer crashes when no text is being passed to the limit text function. [#87797][#87797]
+- In some scenarios, when a DROP INDEX statement was run around the same time as a DROP TABLE or DROP DATABASE statement covering the same data, the `DROP INDEX` garbage-collection job could get caught retrying indefinitely. This has been fixed. [#87721][#87721]
```
Comment (stbof) [inline _includes/releases/v22.2/v22.2.0-alpha.4.md:45]: DROP * should be in backticks.

File: _includes/releases/v22.2/v22.2.0-alpha.4.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+- A consistency check is now skipped/stopped when the collection request is canceled before/while running the check computation. Previously such checks would start and run until completion, and, due to the limited size of the worker pool, prevent the useful checks from running. [#87841][#87841]
+- Consistency checks are now sent to all replicas in parallel. Previously, they were blocked on processing the local replica first. This reduces the latency of one check by 2x and allows better propagation of the cancellation signal, resulting in fewer abandoned tasks on remote replicas and more resources spent on useful checks. [#87841][#87841]
+- Because of a misused query optimization involving tables with one or more PARTITION BY clauses and partition zone constraints which assign region locality to those partitions, in some cases the optimizer would pick a `locality-optimized search` query plan which is not truly locality-optimized and has higher latency than competing query plans which use distributed scan. Locality-optimized search is now avoided in cases which are known not to benefit from this optimization. [#87866][#87866]
```
Comment (stbof) [inline _includes/releases/v22.2/v22.2.0-alpha.4.md:48]: All SQL syntax should be in backticks.

File: _includes/releases/v22.2/v22.2.0-alpha.4.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+<h3 id="v22-2-0-alpha-4-build-changes">Build changes</h3>
+
+- Upgrade to go 1.19.1 [#86049][#86049]
+
+<h3 id="v22-2-0-alpha-4-miscellaneous">Miscellaneous</h3>
+
+<h4 id="v22-2-0-alpha-4-missing-category">Missing category</h4>
+
+- Index recommendation now considers not visible indexes and can also make index recommendations for alter index … visible. [#87174][#87174]
```
Comment (stbof) [inline _includes/releases/v22.2/v22.2.0-alpha.4.md:61]:
```suggestion
- Index recommendation now considers not visible indexes and can also make index recommendations for `ALTER INDEX … VISIBLE`. [#87174][#87174]
```

**Review by stbof — CHANGES_REQUESTED** [review]
Sorry I forgot to start review first.

File: _includes/releases/v22.2/v22.2.0-alpha.4.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+<h3 id="v22-2-0-alpha-4-db-console-changes">DB Console changes</h3>
+
+- Fixed the active transaction description. Removed transaction insights details elapsed time because it is not available and was the contention time. [#87604][#87604]
+- Internal sessions and active executions are now surfaced in the UI only when the cluster setting `sql.stats.response.show_internal.enabled` is set to true. [#87608][#87608]
+- When there is no insight problem detected, a message indicates that the statement was slow and how long it took to execute. [#87799][#87799]
+- The Insight Statement Detail page is linked from the Insight Statement page but no longer includes a link to itself. [#87800][#87800]
+- Added the ability to copy an index recommendation to the clipboard. [#87794][#87794]
+- Fixed the index and grant sorting on the database page to have default column and to have url match the user selection. [#87832][#87832]
+- Added Application Name to Statement overview, Transaction Overview (and their respective column selectors), Transaction Details. Updated label from "App" to "Application Name" on Statement Details page. [#87868][#87868]
```
Comment (stbof) [inline _includes/releases/v22.2/v22.2.0-alpha.4.md:33]:
```suggestion
- Added Application Name to Statement Overview, Transaction Overview (and their respective column selectors), Transaction Details. Updated label from "App" to "Application Name" on Statement Details page. [#87868][#87868]
```

File: _includes/releases/v22.2/v22.2.0-alpha.4.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+- Cockroach demo no longer runs multi-tenant mode by default. [#87810][#87810]
+
+<h3 id="v22-2-0-alpha-4-db-console-changes">DB Console changes</h3>
+
+- Fixed the active transaction description. Removed transaction insights details elapsed time because it is not available and was the contention time. [#87604][#87604]
+- Internal sessions and active executions are now surfaced in the UI only when the cluster setting `sql.stats.response.show_internal.enabled` is set to true. [#87608][#87608]
+- When there is no insight problem detected, a message indicates that the statement was slow and how long it took to execute. [#87799][#87799]
+- The Insight Statement Detail page is linked from the Insight Statement page but no longer includes a link to itself. [#87800][#87800]
+- Added the ability to copy an index recommendation to the clipboard. [#87794][#87794]
+- Fixed the index and grant sorting on the database page to have default column and to have url match the user selection. [#87832][#87832]
```
Comment (stbof) [inline _includes/releases/v22.2/v22.2.0-alpha.4.md:32]:
```suggestion
- Fixed the index and grant sorting on the Databases page to have default column and to have url match the user selection. [#87832][#87832]
```

File: _includes/releases/v22.2/v22.2.0-alpha.4.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+- Added Application Name to Statement overview, Transaction Overview (and their respective column selectors), Transaction Details. Updated label from "App" to "Application Name" on Statement Details page. [#87868][#87868]
+- The value for percentage of all runtime is now calculated based on all data from the time period. Previously, it was calculated based on only the filtered data. [#88027][#88027]
+- The Contention column has been renamed to Contention time. This matches other columns such as Elapsed time. [#88040][#88040]
+- An Insights link has been added to all insights pages, and the message on the Schema Insights message has been updated to match the message on the Workload Insights page. [#88012][#88012]
```
Comment (stbof) [inline _includes/releases/v22.2/v22.2.0-alpha.4.md:36]:
```suggestion
- An Insights link has been added to all insights pages, and the message on the Schema Insights page has been updated to match the message on the Workload Insights page. [#88012][#88012]
```

File: _includes/releases/v22.2/v22.2.0-alpha.4.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+- Instead of always recommending to replace the first existing index with the same explicit columns, the index recommendation now considers all existing indexes in the table to decide the best one to replace. [#87174][#87174]
+- Owners of a backup schedule can now control their schedule using the supported pause, resume, drop, and alter queries. [#87600][#87600]
+- Previously, a schema could be renamed even when a child table was referenced by a view or UDF. Renaming the schema breaks the view or UDF since tables are referenced by name in views and UDFs. A check has been added that disallows renaming a schema that is referenced in views or UDFs. [#87540][#87540]
+- Users may now be granted the `CHANGEFEED` privilege on a table, allowing them to create changefeeds for that table even if they don't have the `CONTROLCHANGEFEED` role option or the `SELECT` privilege. Note that this still in effect gives them the ability to read the data in the table. Users with the `CONTROLCHANGEFEED` role option still need SELECT on each table, even if they also have `CHANGEFEED`. [#87887][#87887]
```
Comment (stbof) [inline _includes/releases/v22.2/v22.2.0-alpha.4.md:18]:
```suggestion
- Users may now be granted the `CHANGEFEED` privilege on a table, allowing them to create changefeeds for that table even if they don't have the `CONTROLCHANGEFEED` role option or the `SELECT` privilege. Note that this still in effect gives them the ability to read the data in the table. Users with the `CONTROLCHANGEFEED` role option still need `SELECT` on each table, even if they also have `CHANGEFEED`. [#87887][#87887]
```

File: _includes/releases/v22.2/v22.2.0-alpha.4.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+<h3 id="v22-2-0-alpha-4-command-line-changes">Command-line changes</h3>
+
+- The \c metacommand no longer shows the password in cleartext. [#87538][#87538]
+- Cockroach demo no longer runs multi-tenant mode by default. [#87810][#87810]
```
Comment (stbof) [inline _includes/releases/v22.2/v22.2.0-alpha.4.md:23]:
```suggestion
- `cockroach demo` no longer runs multi-tenant mode by default. [#87810][#87810]
```

File: _includes/releases/v22.2/v22.2.0-alpha.4.md [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+- Owners of a backup schedule can now control their schedule using the supported pause, resume, drop, and alter queries. [#87600][#87600]
+- Previously, a schema could be renamed even when a child table was referenced by a view or UDF. Renaming the schema breaks the view or UDF since tables are referenced by name in views and UDFs. A check has been added that disallows renaming a schema that is referenced in views or UDFs. [#87540][#87540]
+- Users may now be granted the `CHANGEFEED` privilege on a table, allowing them to create changefeeds for that table even if they don't have the `CONTROLCHANGEFEED` role option or the `SELECT` privilege. Note that this still in effect gives them the ability to read the data in the table. Users with the `CONTROLCHANGEFEED` role option still need SELECT on each table, even if they also have `CHANGEFEED`. [#87887][#87887]
+
+<h3 id="v22-2-0-alpha-4-command-line-changes">Command-line changes</h3>
+
+- The \c metacommand no longer shows the password in cleartext. [#87538][#87538]
```
Comment (stbof) [inline _includes/releases/v22.2/v22.2.0-alpha.4.md:22]:
```suggestion
- The `\c` metacommand no longer shows the password in cleartext. [#87538][#87538]
```

File: _includes/releases/v22.2/v22.2.0-alpha.4.md [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+- Cockroach demo no longer runs multi-tenant mode by default. [#87810][#87810]
+
+<h3 id="v22-2-0-alpha-4-db-console-changes">DB Console changes</h3>
+
+- Fixed the active transaction description. Removed transaction insights details elapsed time because it is not available and was the contention time. [#87604][#87604]
+- Internal sessions and active executions are now surfaced in the UI only when the cluster setting `sql.stats.response.show_internal.enabled` is set to true. [#87608][#87608]
+- When there is no insight problem detected, a message indicates that the statement was slow and how long it took to execute. [#87799][#87799]
+- The Insight Statement Detail page is linked from the Insight Statement page but no longer includes a link to itself. [#87800][#87800]
+- Added the ability to copy an index recommendation to the clipboard. [#87794][#87794]
+- Fixed the index and grant sorting on the Database page to have default column and to have url match the user selection. [#87832][#87832]
```
Comment (stbof) [inline _includes/releases/v22.2/v22.2.0-alpha.4.md:32]:
```suggestion
- Fixed the index and grant sorting on the Databases page to have default column and to have URL match the user selection. [#87832][#87832]
```

**Review by stbof — APPROVED** [review]
Approved after suggestions are committed.

File: _includes/releases/v22.2/v22.2.0-alpha.4.md [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+- Owners of a backup schedule can now control their schedule using the supported pause, resume, drop, and alter queries. [#87600][#87600]
+- Previously, a schema could be renamed even when a child table was referenced by a view or UDF. Renaming the schema breaks the view or UDF since tables are referenced by name in views and UDFs. A check has been added that disallows renaming a schema that is referenced in views or UDFs. [#87540][#87540]
+- Users may now be granted the `CHANGEFEED` privilege on a table, allowing them to create changefeeds for that table even if they don't have the `CONTROLCHANGEFEED` role option or the `SELECT` privilege. Note that this still in effect gives them the ability to read the data in the table. Users with the `CONTROLCHANGEFEED` role option still need `SELECT` on each table, even if they also have `CHANGEFEED`. [#87887][#87887]
+
+<h3 id="v22-2-0-alpha-4-command-line-changes">Command-line changes</h3>
+
+- The `\c` metacommand no longer shows the password in cleartext. [#87538][#87538]
+- `cockroach demo` no longer runs multi-tenant mode by default. [#87810][#87810]
```
Comment (stbof) [inline _includes/releases/v22.2/v22.2.0-alpha.4.md:23]: I remembered that we aren't supposed to talk about `multi-tenant` mode in release notes. Perhaps should remove this.

Comment (mdlinville) [reply _includes/releases/v22.2/v22.2.0-alpha.4.md:23]: I wondered about that, but I also wondered if this was more about `cockroach demo` having surprising behavior for a bit and not having it anymore. I'll find out whether I should remove it or not.

Comment (mdlinville) [reply _includes/releases/v22.2/v22.2.0-alpha.4.md]: Confirmed it can be removed.

**Pull request merged**